### PR TITLE
Rebase 4343

### PR DESCRIPTION
**Pull request by bmac**

### CODE REVIEW

File: addon/index.js [hunk 1]
```diff
@@ -163,6 +166,18 @@ Object.defineProperty(DS, 'normalizeModelName', {
   value: normalizeModelName
 });
 
-Ember.lookup.DS = DS;
+Object.defineProperty(global, 'DS', {
+  configurable: true,
+  get() {
+    deprecate(
+      'Using the global version of DS is deprecated. Please either import ' +
+        'the specific modules needed or `import DS from \'ember-data\';`.',
```
Comment (pangratz) [inline addon/index.js]: Tiny nitpick: do we want to advertise the `import DS from 'ember.data'`? I think when a pre-addonified-version of ember-data is used, with the shims all the necessary imports should be available via modules, so there is no need for importing `DS` namespace?

**Comment by pangratz** [comment]
LGTM 👍

File: addon/index.js (hunk 1 above)
Comment (bmac) [inline addon/index.js]: I think suggesting users import `DS` is the easiest fix for them if they are using the `DS` global.

Comment (pangratz) [inline addon/index.js]: You're right!

**Pull request merged**